packages/nimble-components/src/table/models/virtualizer.ts

```diff
@@ -110,4 +113,31 @@ export class Virtualizer<TData extends TableRecord = TableRecord> {
 
         this.rowContainerYOffset = rowContainerYOffset;
     }
+
+    private notifyFocusedCellRecycling(): void {
+        let tableFocusedElement = this.table.shadowRoot!.activeElement;
+        if (tableFocusedElement) {
+            while (
+                tableFocusedElement !== null
+                && !(tableFocusedElement instanceof TableCellView)
+            ) {
+                if (tableFocusedElement.shadowRoot) {
+                    tableFocusedElement = tableFocusedElement.shadowRoot.activeElement;
+                } else {
+                    break;
+                }
+            }
+            if (tableFocusedElement instanceof TableCellView) {
+                tableFocusedElement.focusedRecycleCallback();
+            }
+        }
+        if (this.table.openActionMenuRecordId !== undefined) {
+            const row = Array.prototype.find.call(
```

Comment:
Since we didn't fully get consensus on the best way to implement closing action menus in the HLD PR, figured I'd leave an open comment here for discussion. Potential options, other than what I have here:
- `querySelectorAll` instead of iterating the row container children, and similarly in the row code (i.e. `querySelectorAll('nimble-table-row[record-id=...]')`, `row.shadowRoot.querySelectorAll('nimble-table-cell[menu-open]')` - this is closer to what I had in the prototype
- Storing off a reference (`WeakRef`?) to the row and cell DOM elements when an action menu is open (store ref in `beforetoggle`, clear refs in `toggle` when menu is transitioning to closed), and use that to get back to the menu button

Reply:
I think I'm fine with the approach you have here.

Reply:
Maybe we can use the [children directive](https://www.fast.design/docs/fast-element/using-directives#the-children-directive)?

```ts
`<div class="table-row-container" ${children('rowElements'} role="rowgroup">`
```

Then can just do `this.table.rowElements.find(row => row.recordId === this.table.openActionMenuRecordId)`

Looks like there are only rows so don't really need to filter but the directive has filter settings if needed.

Reply:
Updated to use children directive